contrib/openapi/spec.yml

```diff
@@ -1,453 +1,282 @@
 swagger: '2.0'
+schemes:
+  - http
+host: api.xsnippet.org
 info:
   title: XSnippet API
-  description: REST API of XSnippet, a simple web-service for sharing code snippets
-  version: "4.0.0"
+  version: '1.0'
+  contact:
+    email: dev@xsnippet.org
+  description: |
+    The XSnippet API is organized around [REST]. Our API has predictable,
+    resource-oriented URLs, and uses HTTP response codes to indicate API
+    errors. We use built-in HTTP features, like HTTP authentication and
+    HTTP verbs, which are understood by off-the-shelf HTTP clients. See
+    more details below and feel free to contact us if any questions!
 
-host: api.xsnippet.org
+    [REST]: https://en.wikipedia.org/wiki/Representational_state_transfer
 
-schemes:
-  - http
-  - https
+    Authentication
+    ==============
+
+    Authentication is done via [JSON Web Token] (JWT) that is issued by the
+    API in exchange for Google Access Token. The token needs to be passed
+    via `Authorization` HTTP header, and is mandatory for those endpoints
+    which require non-anonymous permissions.
+
+    ```http
+    Authorization: bearer {your-token-here}
+    ```
+
+    [JSON Web Token]: https://tools.ietf.org/html/rfc7519
+
+    Pagination
+    ==========
 
+    Pagination is done via two query parameters: `limit` and `marker`. The
+    first one limits a number of entries to be returned in response, while
+    the last one – defines your position in the list by taking an entry ID
+    and returning `limit` entries after that one assuming that entries are
+    sorted in some (defined by concrete API endpoint) order.
+
+    In other words if you make a list request and receive `100` entries,
+    ending with `foo`, your subsequent call can include `?marker=foo` in
+    order to fetch the next page of the list.
+
+    Versioning
+    ==========
+
+    API version is passed via non-standard `Api-Version` HTTP header, and
+    must be equal to `1.0` so far. In case the version is not passed, the
```

Comment:
I'm not sure that defaulting to oldest version is a good idea either. You see if we default to oldest version then we are in the situation when we must support this version forever. Otherwise, we will break users anyway. Maybe the best option here is to force users to pass a version, but it will complicate API consuming via `curl` and company.

Nevertheless I just documented what we have right now. Could you please create a corresponding GitHub issue so we can move discussion and implementation there?